Correct the result if no module version is verified

juhlee-microsoft opened the pull request:
Depending on the distro and its module loading state, the test result should be accurate. The old code shows 'passed' all the times regardless LIS module version check or not. 

The change will filter out no module version check case, and turn in 'failed' result, if not.

**Before changing**
```
[SUSE SLES 15 latest] [LISAv2 Test Results Summary]
[SUSE SLES 15 latest] Test Run On           : 12/06/2019 22:25:04
[SUSE SLES 15 latest] ARM Image Under Test  : SUSE : SLES : 15 : latest
[SUSE SLES 15 latest] Initial Kernel Version: 4.12.14-5.30-azure
[SUSE SLES 15 latest] Final Kernel Version  : 4.12.14-5.30-azure
[SUSE SLES 15 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[SUSE SLES 15 latest] Total Time (dd:hh:mm) : 0:0:4
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[SUSE SLES 15 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[SUSE SLES 15 latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                 0.46 
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest] Logs can be found at C:\LISAv2\CH93\TestResults\2019-06-12-22-25-00-9426
```
In side of TestExecution.log,
```
Fri Dec 06 22:29:00 2019 : Target module is empty or null
Fri Dec 06 22:29:00 2019 : RPM availability in the system: true
Fri Dec 06 22:29:00 2019 : The current LIS installation state: 
Fri Dec 06 22:29:00 2019 : Exiting with state: **TestCompleted**.
```
**After changing**
```
[SUSE SLES 15 latest] [LISAv2 Test Results Summary]
[SUSE SLES 15 latest] Test Run On           : 12/06/2019 22:50:55
[SUSE SLES 15 latest] ARM Image Under Test  : SUSE : SLES : 15 : latest
[SUSE SLES 15 latest] Initial Kernel Version: 4.12.14-5.30-azure
[SUSE SLES 15 latest] Final Kernel Version  : 4.12.14-5.30-azure
[SUSE SLES 15 latest] Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
[SUSE SLES 15 latest] Total Time (dd:hh:mm) : 0:0:4
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[SUSE SLES 15 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[SUSE SLES 15 latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        **FAIL**                 1.47 
[SUSE SLES 15 latest] Calling function - Run-LinuxCmd. Failed to execute : bash VERIFY-LIS-MODULES-VERSION.sh > VERIFY-LIS-MODULES-VERSION_summary.log 2>&1.
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest] Logs can be found at C:\LISAv2\VO24\TestResults\2019-06-12-22-50-51-2120

[Canonical UbuntuServer 12.04.5-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 12.04.5-LTS latest] Test Run On           : 12/06/2019 22:36:12
[Canonical UbuntuServer 12.04.5-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 12.04.5-LTS : latest
[Canonical UbuntuServer 12.04.5-LTS latest] Initial Kernel Version: 3.13.0-117-generic
[Canonical UbuntuServer 12.04.5-LTS latest] Final Kernel Version  : 3.13.0-117-generic
[Canonical UbuntuServer 12.04.5-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 12.04.5-LTS latest] Total Time (dd:hh:mm) : 0:0:2
[Canonical UbuntuServer 12.04.5-LTS latest] 
[Canonical UbuntuServer 12.04.5-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 12.04.5-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 12.04.5-LTS latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                 0.42 
[Canonical UbuntuServer 12.04.5-LTS latest] 
[Canonical UbuntuServer 12.04.5-LTS latest] 
[Canonical UbuntuServer 12.04.5-LTS latest] Logs can be found at C:\LISAv2\OQ27\TestResults\2019-06-12-22-36-08-3330
```
They are also regression result with other distros
```
[OpenLogic CentOS 7.1 latest] [LISAv2 Test Results Summary]
[OpenLogic CentOS 7.1 latest] Test Run On           : 12/06/2019 22:36:46
[OpenLogic CentOS 7.1 latest] ARM Image Under Test  : OpenLogic : CentOS : 7.1 : latest
[OpenLogic CentOS 7.1 latest] Initial Kernel Version: 3.10.0-229.20.1.el7.x86_64
[OpenLogic CentOS 7.1 latest] Final Kernel Version  : 3.10.0-229.20.1.el7.x86_64
[OpenLogic CentOS 7.1 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[OpenLogic CentOS 7.1 latest] Total Time (dd:hh:mm) : 0:0:2
[OpenLogic CentOS 7.1 latest] 
[OpenLogic CentOS 7.1 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[OpenLogic CentOS 7.1 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[OpenLogic CentOS 7.1 latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                 0.48 
[OpenLogic CentOS 7.1 latest] 
[OpenLogic CentOS 7.1 latest] 
[OpenLogic CentOS 7.1 latest] Logs can be found at C:\LISAv2\AD45-20191206223639\TestResults\2019-06-12-22-36-42-2320

[RedHat RHEL 7.7 latest] [LISAv2 Test Results Summary]
[RedHat RHEL 7.7 latest] Test Run On           : 12/06/2019 22:36:40
[RedHat RHEL 7.7 latest] ARM Image Under Test  : RedHat : RHEL : 7.7 : latest
[RedHat RHEL 7.7 latest] Initial Kernel Version: 3.10.0-1062.el7.x86_64
[RedHat RHEL 7.7 latest] Final Kernel Version  : 3.10.0-1062.el7.x86_64
[RedHat RHEL 7.7 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[RedHat RHEL 7.7 latest] Total Time (dd:hh:mm) : 0:0:3
[RedHat RHEL 7.7 latest] 
[RedHat RHEL 7.7 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[RedHat RHEL 7.7 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[RedHat RHEL 7.7 latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                 0.82 
[RedHat RHEL 7.7 latest] 
[RedHat RHEL 7.7 latest] 
[RedHat RHEL 7.7 latest] Logs can be found at C:\LISAv2\DR49\TestResults\2019-06-12-22-36-34-0509


[Canonical UbuntuServer 19.04 latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 19.04 latest] Test Run On           : 12/06/2019 22:36:24
[Canonical UbuntuServer 19.04 latest] ARM Image Under Test  : Canonical : UbuntuServer : 19.04 : latest
[Canonical UbuntuServer 19.04 latest] Initial Kernel Version: 5.0.0-1025-azure
[Canonical UbuntuServer 19.04 latest] Final Kernel Version  : 5.0.0-1025-azure
[Canonical UbuntuServer 19.04 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 19.04 latest] Total Time (dd:hh:mm) : 0:0:3
[Canonical UbuntuServer 19.04 latest] 
[Canonical UbuntuServer 19.04 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 19.04 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 19.04 latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                 0.53 
[Canonical UbuntuServer 19.04 latest] 
[Canonical UbuntuServer 19.04 latest] 
[Canonical UbuntuServer 19.04 latest] Logs can be found at C:\LISAv2\QJ35\TestResults\2019-06-12-22-36-20-0458

[OpenLogic CentOS 7.6 latest] [LISAv2 Test Results Summary]
[OpenLogic CentOS 7.6 latest] Test Run On           : 12/06/2019 22:36:49
[OpenLogic CentOS 7.6 latest] ARM Image Under Test  : OpenLogic : CentOS : 7.6 : latest
[OpenLogic CentOS 7.6 latest] Initial Kernel Version: 3.10.0-957.27.2.el7.x86_64
[OpenLogic CentOS 7.6 latest] Final Kernel Version  : 3.10.0-957.27.2.el7.x86_64
[OpenLogic CentOS 7.6 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[OpenLogic CentOS 7.6 latest] Total Time (dd:hh:mm) : 0:0:3
[OpenLogic CentOS 7.6 latest] 
[OpenLogic CentOS 7.6 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[OpenLogic CentOS 7.6 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[OpenLogic CentOS 7.6 latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                  0.8 
[OpenLogic CentOS 7.6 latest] 
[OpenLogic CentOS 7.6 latest] 
[OpenLogic CentOS 7.6 latest] Logs can be found at C:\LISAv2\YR26\TestResults\2019-06-12-22-36-45-3650

[Canonical UbuntuServer 18.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 18.04-LTS latest] Test Run On           : 12/06/2019 22:36:21
[Canonical UbuntuServer 18.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[Canonical UbuntuServer 18.04-LTS latest] Initial Kernel Version: 5.0.0-1027-azure
[Canonical UbuntuServer 18.04-LTS latest] Final Kernel Version  : 5.0.0-1027-azure
[Canonical UbuntuServer 18.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 18.04-LTS latest] Total Time (dd:hh:mm) : 0:0:3
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 18.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 18.04-LTS latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                  0.8 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] 
[Canonical UbuntuServer 18.04-LTS latest] Logs can be found at C:\LISAv2\SJ22\TestResults\2019-06-12-22-36-15-8948

[RedHat RHEL 6.9 latest] [LISAv2 Test Results Summary]
[RedHat RHEL 6.9 latest] Test Run On           : 12/06/2019 22:36:34
[RedHat RHEL 6.9 latest] ARM Image Under Test  : RedHat : RHEL : 6.9 : latest
[RedHat RHEL 6.9 latest] Initial Kernel Version: 2.6.32-696.18.7.el6.x86_64
[RedHat RHEL 6.9 latest] Final Kernel Version  : 2.6.32-696.18.7.el6.x86_64
[RedHat RHEL 6.9 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[RedHat RHEL 6.9 latest] Total Time (dd:hh:mm) : 0:0:3
[RedHat RHEL 6.9 latest] 
[RedHat RHEL 6.9 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[RedHat RHEL 6.9 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[RedHat RHEL 6.9 latest]     1 CORE                 VERIFY-LIS-MODULES-VERSION                                                        PASS                 0.66 
[RedHat RHEL 6.9 latest] 
[RedHat RHEL 6.9 latest] 
[RedHat RHEL 6.9 latest] Logs can be found at C:\LISAv2\SO40-20191206223627\TestResults\2019-06-12-22-36-29-9279
```

LiliDeng commented:
This is for testing CommonFunctions.psm1 not break other distro.

Test Run On           : 01/27/2019 14:17:32
ARM Image Under Test  : credativ : Debian : 9-backports : Latest
Test Category         : Functional
Test Area             : SRIOV
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:12

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 VERIFY-SRIOV-LSPCI                                                                PASS                 3.22

LiliDeng commented:
@Tong-Xu Please run some testing for DPDK, we hit some issue which actual kernel support, but code logic detect it not support when compare kernel version, like RedHat RHEL 7.5 Latest, OpenLogic CentOS 7.5 Latest and SUSE SLES-Standard 15 Latest. And post results here.

LiliDeng commented:
From Tong, before test cases will not run since it didn't pass the kernel version check.

![image](https://user-images.githubusercontent.com/10083705/51824373-05cf3480-231d-11e9-984a-ab84079feba5.png)

**Investigating** fail and abort on CentOS 7.5 and SLES 15

LiliDeng commented:
Reason found, build dpdk failed, since the kernel-devel version is not matched current kernel version, either install matched kernel-devel rpm or update kernel (need reboot)

LiliDeng commented on Testscripts/Windows/VERIFY-DPDK-COMPLIANCE.ps1:
```diff
@@ -5,38 +5,68 @@ param([object] $AllVmData,
-			Write-LogErr $msg
-			throw $msg
+		switch ( $detectedDistro ) {
+			"UBUNTU" {
```
Before updated, $UbuntuSupportKernelVersion is empty.
if ($currentKernelVersion -ge $UbuntuSupportKernelVersion) {

LiliDeng replied:
Let me update using dictionary.

LiliDeng commented:
@Tong-Xu Could you please run testing against my latest code? Run different distro to make sure no regression. Thanks!

ader1990 commented:
I see you refactored these scripts as I did for DPDK in  https://github.com/LIS/LISAv2/pull/621/files#diff-e94b392cb16d037f30600043594c76e6R10

Can we reuse this code for all the dpdk scripts somehow?

LiliDeng commented:
canonical ubuntuserver 18.04-lts Latest
  1 VERIFY-DPDK-FAILSAFE-DURING-TRAFFIC                PASS                29.29
  2 VERIFY-DPDK-BUILD-AND-TESTPMD-TEST                   PASS                25.98
  3 VERIFY-SRIOV-FAILSAFE-FOR-DPDK                              PASS                42.85
  4 VERIFY-DPDK-COMPLIANCE                                  PASS            9.94 

RedHat RHEL 7.5 Latest
  1 VERIFY-DPDK-FAILSAFE-DURING-TRAFFIC                 PASS                32.64 
  2 VERIFY-DPDK-BUILD-AND-TESTPMD-TEST                  PASS                35.83   
  3 VERIFY-SRIOV-FAILSAFE-FOR-DPDK                              PASS            58.5
  4 VERIFY-DPDK-COMPLIANCE                                  PASS                14.09 

OpenLogic CentOS 7.5 Latest
  1 VERIFY-DPDK-FAILSAFE-DURING-TRAFFIC             PASS                32.56      
  2 VERIFY-DPDK-BUILD-AND-TESTPMD-TEST           PASS                 33.1    
  3 VERIFY-SRIOV-FAILSAFE-FOR-DPDK                         PASS                51.93  
  4 VERIFY-DPDK-COMPLIANCE                                  PASS                 13.3 

SUSE SLES-Standard 15 Latest
  1 VERIFY-DPDK-FAILSAFE-DURING-TRAFFIC           PASS                29.98
  2 VERIFY-DPDK-BUILD-AND-TESTPMD-TEST          PASS                26.06        
  3 VERIFY-SRIOV-FAILSAFE-FOR-DPDK                      PASS                47.93 
  4 VERIFY-DPDK-COMPLIANCE                              PASS                10.48

LiliDeng commented:
Reach the agreement with @ader1990 using Teams.

LiliDeng commented:
@lizzha Done for testing, please review again, thanks!

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

Update VERIFY-LIS-MODULES-VERSION.sh

Update VERIFY-LIS-MODULES-VERSION.sh

Update VERIFY-LIS-MODULES-VERSION.sh

Update VERIFY-LIS-MODULES-VERSION.sh

Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Merge remote-tracking branch 'origin/juhlee/state_param' into juhlee/state_param

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh

juhlee-microsoft committed:
Added another filter if non CentOS or RHEL distro

It will be test skipped.

juhlee-microsoft committed:
Update VERIFY-LIS-MODULES-VERSION.sh
